lib/output.js

```diff
@@ -1964,19 +1964,14 @@ function OutputStream(options) {
     function make_num(num) {
         var str = num.toString(10), a = [ str.replace(/^0\./, ".").replace("e+", "e") ], m;
         if (Math.floor(num) === num) {
-            if (num >= 0) {
-                a.push("0x" + num.toString(16).toLowerCase(), // probably pointless
-                       "0" + num.toString(8)); // same.
```

Comment:
Good catch! [Octal literals are deprecated](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Deprecated_octal) and throw a `SyntaxError` in strict mode, so we should never consider using them. Fortunately, as the comment points out, they probably never led to shorter literals than decimal or hexadecimal literals.

Reply:
One more point is that octal literals are never shorter then dec or hex